Remove DC/OS E2E

bamarni opened the pull request:
Currently we have a mix of DC/OS E2E and DC/OS Launch. We pinned an old
version of DC/OS E2E which doesn't work with DC/OS dev anymore, updating
DC/OS E2E to its latest version would probably fix the issue, this was
tried in #376 without success.

As a simple fix we can remove DC/OS E2E and stick to DC/OS Launch for
now, in the long-term we should move to the universal DC/OS installer.

bamarni commented:
@mesosphere-ci run all integration tests

bamarni commented:
@mesosphere-ci run linux integration tests

bamarni commented:
@mesosphere-ci run linux integration tests

bamarni commented:
@mesosphere-ci run linux integration tests

bamarni commented:
@mesosphere-ci run mac integration tests

bamarni commented:
@mesosphere-ci run linux integration tests

bamarni commented:
@mesosphere-ci run mac integration tests

bamarni commented:
@mesosphere-ci run linux integration tests

bamarni commented:
@mesosphere-ci run mac integration tests

bamarni commented:
@mesosphere-ci run linux integration tests

bamarni commented:
@mesosphere-ci run mac integration tests

bamarni commented:
@mesosphere-ci run linux integration tests

bamarni commented:
@mesosphere-ci run mac integration tests

bamarni commented:
@mesosphere-ci run linux integration tests

bamarni commented:
@mesosphere-ci run mac integration tests

bamarni commented:
@mesosphere-ci run linux integration tests

bamarni commented:
@mesosphere-ci run mac integration tests

drewvanstone commented:
Looks like the build is failing

bamarni commented:
It passed on linux but there is an escape issue on mac with the jenkinsfile, looking at it.

bamarni committed:
Remove DC/OS E2E

Currently we have a mix of DC/OS E2E and DC/OS Launch. We pinned an old
version of DC/OS E2E which doesn't work with DC/OS dev anymore, updating
DC/OS E2E to its latest version would probably fix the issue, this was
tried in #376 without success.

As a simple fix we can remove DC/OS E2E and stick to DC/OS Launch for
now, in the long-term we should move to the universal DC/OS installer.

bamarni commented:
@mesosphere-ci run linux integration tests

bamarni commented:
@mesosphere-ci run mac integration tests

bamarni commented:
@mesosphere-ci run windows integration tests

janisz commented:
I'm not a fan of this change. Instead I'd like to invest more time and fix what was broken.

bamarni commented:
@mesosphere-ci run windows integration tests

janisz reviewed (changes requested):
I need to block this PR. IMO dcos-launch is a problem not a solution. We should stay away from deprecated software.

bamarni committed:
Use a stable error message for os.Stat

This makes it easier to assert against, we still log the low-lever error
which would be printed with the `-vv` verbose flags.

bamarni commented:
@mesosphere-ci run windows integration tests

janisz commented on scripts/requirements.txt:
```diff
@@ -1,11 +1,9 @@
-dcos-e2e==2019.04.02.1
 dcos_launch==0.1-dev
 dcos-test-utils==0.1
+passlib==1.7.1
```
Why new dependency?

bamarni replied:
We're relying on this dependency in the `launch_aws_cluster.py` script but it was missing here, it was still getting installed as it's an indirect dependency (through dcos-e2e).

bamarni commented:
@mesosphere-ci run linux integration tests

bamarni commented:
@mesosphere-ci run mac integration tests

makkes commented:
@mesosphere-ci run mac integration tests

makkes commented:
@mesosphere-ci run windows integration tests

makkes commented:
I had to merge `2.1-patch.x` in in order to be able to merge this.

bamarni commented:
I'll rebase to keep the history clean and remove the extra lines in Go

bamarni commented:
@mesosphere-ci run all integration tests

bamarni commented:
I've canceled the windows build as the previous one already passed, it seemed stuck trying to spin-up the cluster (we can track if it happens again in https://jira.mesosphere.com/browse/DCOS_OSS-5614).
